File: src/resources/ThroughputConnectionSettings.ts
export default interface ThroughputConnectionSettings {
  service_control_settings: ThroughputConnectionSetting[];
  monitoring_settings: ThroughputConnectionSetting[];
  broker_settings: ThroughputConnectionSetting[];
}

export interface ThroughputConnectionSetting {
  name: string;
  description: string;
}

File: src/Particular.LicensingComponent.Contracts/ThroughputConnectionSettings.cs
namespace Particular.LicensingComponent.Contracts;

public class ThroughputConnectionSettings
{
    public List<ThroughputConnectionSetting> ServiceControlSettings { get; set; } = [];
    public List<ThroughputConnectionSetting> MonitoringSettings { get; set; } = [];
    public List<ThroughputConnectionSetting> BrokerSettings { get; set; } = [];
}

public class ThroughputConnectionSetting(string name, string description)
{
    public string Name { get; } = name;
    public string Description { get; } = description;
}